spec/command_spec.rb

```diff
@@ -0,0 +1,115 @@
+require 'gorgon/command'
+
+describe Gorgon::Command do
+  context 'start' do
+    it 'starts the test suite' do
+      originator = double('originator')
+      originator.should_receive(:originate).with(no_args).and_return(0)
+      Originator.should_receive(:new).with(no_args).and_return(originator)
+      begin
+        Gorgon::Command.new([]).run!('start')
+      rescue SystemExit => e
+        error_code = e.status
+      end
+
+      error_code.should eq(0)
+    end
+  end
+
+  context 'listen' do
+    it 'starts listener' do
+      listener = double('listener')
+      listener.should_receive(:listen).with(no_args).and_return(true)
+      Listener.should_receive(:new).with(no_args).and_return(listener)
+      Gorgon::Command.new([]).run!('listen')
+    end
+  end
+
+  context 'start rsync' do
+    it 'starts rsync for directory' do
+      RsyncDaemon.should_receive(:start).with('/path/to/directory').and_return(true)
+      Gorgon::Command.new(['/path/to/directory']).run!('start_rsync')
```

Comment:
The `Gorgon::Command`'s API could be improved. It's strange that we are passing the options of the command to the constructor and the command to the `run` method, when the command's options depend on the command. I would rather expose a class method `Gorgon::Command.run(command, options)` or even just `Gorgon::Command.run(argv)` and let the `run` class method create the instance and call the instance method. That would make the API simpler.
It would even allow us to test [this code](https://github.com/nulogy/Gorgon/pull/78/files#diff-8677d8efe3d3823cfeaa98b311f667a3R4).